language/functional_tests/tests/testsuite/borrow_tests/factor_invalid_1.mvir

```diff
@@ -0,0 +1,52 @@
+module M {
+    struct X { f: Self.Y }
+    struct Y { g: u64, h: u64 }
+
+    t1() {
+        let x: Self.X;
+        let ref_x: &mut Self.X;
+        let ref_x_f: &mut Self.Y;
+        let ref_x_f_g: &mut u64;
+        let ref_x_f_h: &mut u64;
+
+        x = X { f: Y { g: 0, h: 0 } };
+
+        // ref_x_f_g = &mut x.f.g
+        ref_x = &mut x;
+        ref_x_f = &mut move(ref_x).f;
+        ref_x_f_g = &mut copy(ref_x_f).g;
+
+        Self.foo(move(ref_x_f_g), move(ref_x_f));
+        return;
+    }
+
+    t2() {
+        let x: Self.X;
+        let ref_x: &mut Self.X;
+        let ref_x_f: &mut Self.Y;
+        let ref_x_f_g: &mut u64;
+        let ref_x_f_h: &mut u64;
+
+        x = X { f: Y { g: 0, h: 0 } };
+
+        // ref_x_f_g = &mut x.f.g
+        ref_x = &mut x;
+        ref_x_f = &mut move(ref_x).f;
+        ref_x_f_g = &mut copy(ref_x_f).g;
+
+        Self.bar(move(ref_x_f), move(ref_x_f_g));
+        return;
+    }
+
+    foo(a: &mut u64, b: &mut Self.Y) {
+        return;
+    }
+
+    bar(a: &mut Self.Y, b: &mut u64) {
+        return;
+    }
+}
+
+// check: VerificationFailure
+// check: CALL_BORROWED_MUTABLE_REFERENCE_ERROR
```

Comment:
nit could you comment where these errors occur?

Reply:
Done.